src/utils/hmac-validator.ts

```diff
@@ -36,8 +36,8 @@ export default function validateHmac(query: AuthQuery): boolean {
       'Query does not contain an HMAC value.',
     );
   }
-  const {hmac, code, timestamp, state, shop} = query;
-  const localHmac = generateLocalHmac({code, timestamp, state, shop});
+  const {hmac, code, timestamp, state, shop, host} = query;
+  const localHmac = generateLocalHmac({code, timestamp, state, shop, ...host && {host}});
```

Comment:
I also realized that we might want to do this in `generateLocalHmac` instead, to make sure we're always filtering out the keys we don't want.